sdk/rust/src/pvrecorder.rs

```diff
@@ -339,27 +365,22 @@ impl RecorderInner {
         }
 
         let status = unsafe { (self.vtable.pv_recorder_read)(self.cpvrecorder, pcm.as_mut_ptr()) };
-        check_fn_call_status!(status, "pv_recorder_read");
-
-        return Ok(());
+        check_fn_call_status(status, "pv_recorder_read")
     }
 
-    pub fn get_audio_devices() -> Result<Vec<String>, RecorderError> {
+    pub fn get_audio_devices(&self) -> Result<Vec<String>, RecorderError> {
```

Comment:
Signature change here.

Because we don't rely on the global library anymore, we need `&self` for the vtable